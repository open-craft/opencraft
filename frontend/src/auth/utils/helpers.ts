import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
