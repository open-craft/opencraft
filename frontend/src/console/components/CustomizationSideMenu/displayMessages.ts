const messages = {
  accordionTheme: {
    defaultMessage: 'Theme',
    description: 'The name of the theme menu in the sidebar accordion.'
  },
  linkPreviewColors: {
    defaultMessage: 'Preview & colors',
    description: 'Menu name.'
  },
  linkLogos: {
    defaultMessage: 'Logos',
    description: 'Menu name.'
  },
  linkButtons: {
    defaultMessage: 'Buttons',
    description: 'Menu name.'
  },
  linkNavigation: {
    defaultMessage: 'Navigation',
    description: 'Menu name.'
  },
  accordionCustomPages: {
    defaultMessage: 'Custom Pages',
    description: 'The name of the custom page menu in the sidebar accordion.'
  },
  linkAbout: {
    defaultMessage: 'About',
    description: 'Menu name.'
  },
  linkTOS: {
    defaultMessage: 'Terms of Service',
    description: 'Menu name.'
  },
  linkContact: {
    defaultMessage: 'Contact',
    description: 'Menu name.'
  },
  accordionInstanceSettings: {
    defaultMessage: 'Instance Settings',
    description: 'The name of the custom page menu in the sidebar accordion.'
  },
  linkGeneral: {
    defaultMessage: 'General',
    description: 'The name of the custom page menu in the sidebar accordion.'
  },
  linkDomain: {
    defaultMessage: 'Domain',
    description: 'The name of the custom page menu in the sidebar accordion.'
  }
};

export default messages;
