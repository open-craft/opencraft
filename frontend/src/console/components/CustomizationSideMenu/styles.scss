@import "~styles/theme";


.customization-menu {
  padding-bottom: 30px;

  .card {
    border: none;

    .card-header {
      background-color: white;
      border: none;
      padding: 5px 15px 5px;
      font-family: $chivo-font;
      font-size: 22px;
      letter-spacing: 0.66px;
      color: #0f1f24;

      &:hover {
        text-decoration: underline;
      }
    }

    .card-body {
      padding: 0 15px 0;

      a {
        padding: 10px 15px;
        border-top: 1px solid $light-gray;
        font-family: $chivo-font;
        font-size: 18px;
        letter-spacing: 1px;
        color: $secondary-1;

        &:hover, &:focus {
          text-decoration: underline;
        }

        &.disabled {
          color: $secondary-2;
          text-decoration: none;
          cursor: default;
        }

        &.active {
          background-color: $light-blue-2;
        }
      }
    }
  }
}
