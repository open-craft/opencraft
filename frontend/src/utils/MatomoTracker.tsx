import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
