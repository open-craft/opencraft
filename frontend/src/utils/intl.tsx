import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
