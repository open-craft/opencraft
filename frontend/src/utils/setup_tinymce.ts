import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
