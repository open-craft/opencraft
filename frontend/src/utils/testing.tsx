import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
