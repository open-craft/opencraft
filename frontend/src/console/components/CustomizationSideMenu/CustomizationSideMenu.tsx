import * as React from 'react';
import { WrappedMessage } from 'utils/intl';
import { Accordion, Card, Nav } from 'react-bootstrap';
import { ROUTES, AVAILABLE_CUSTOM_PAGES } from 'global/constants';
import './styles.scss';
import { useLocation, NavLink } from 'react-router-dom';
import { capitalizeFirstLetter } from 'utils/string_utils';
import messages from './displayMessages';

export const CustomizationSideMenu: React.FC = () => {
  // Using react hooks to fetch full path and highlight currently active
  // page and extend correct accordion.
  const currentLocation = useLocation().pathname;

  let activeKey = 0;
  if (currentLocation.includes('custom-pages')) {
    activeKey = 1;
  } else if (currentLocation.includes('settings')) {
    activeKey = 2;
  }

  const customPageLink = (pageName: string) => {
    const pageRoute = ROUTES.Console.CUSTOM_PAGES.replace(
      ':pageName',
      pageName
    );
    const intlString = `customPage${capitalizeFirstLetter(pageName)}`;

    return (
      <NavLink exact to={pageRoute} key={`static_page_${pageName}`}>
        <WrappedMessage messages={messages} id={intlString} />
      </NavLink>
    );
  };

  return (
    <Accordion defaultActiveKey={`${activeKey}`} className="customization-menu">
      <Card>
        <Card.Header>
          <Accordion.Toggle as={Card.Header} variant="link" eventKey="0">
            <WrappedMessage messages={messages} id="accordionTheme" />
          </Accordion.Toggle>
        </Card.Header>
        <Accordion.Collapse eventKey="0">
          <Card.Body>
            <Nav className="flex-column">
              <NavLink exact to={ROUTES.Console.THEME_PREVIEW_AND_COLORS}>
                <WrappedMessage messages={messages} id="linkPreviewColors" />
              </NavLink>
              <NavLink exact to={ROUTES.Console.LOGOS}>
                <WrappedMessage messages={messages} id="linkLogos" />
              </NavLink>
              <NavLink exact to={ROUTES.Console.THEME_BUTTONS}>
                <WrappedMessage messages={messages} id="linkButtons" />
              </NavLink>
              <NavLink exact to={ROUTES.Console.THEME_NAVIGATION}>
                <WrappedMessage messages={messages} id="linkNavigation" />
              </NavLink>
              <NavLink exact to={ROUTES.Console.THEME_FOOTER}>
                <WrappedMessage messages={messages} id="linkFooter" />
              </NavLink>
              <NavLink exact to={ROUTES.Console.HERO}>
                <WrappedMessage messages={messages} id="linkHero" />
              </NavLink>
            </Nav>
          </Card.Body>
        </Accordion.Collapse>
      </Card>
      <Card>
        <Card.Header>
          <Accordion.Toggle as={Card.Header} variant="link" eventKey="1">
            <WrappedMessage messages={messages} id="accordionCustomPages" />
          </Accordion.Toggle>
        </Card.Header>
        <Accordion.Collapse eventKey="1">
          <Card.Body>
            <Nav className="flex-column">
              {AVAILABLE_CUSTOM_PAGES.map((pageName) => {
                return customPageLink(pageName);
              })}
            </Nav>
          </Card.Body>
        </Accordion.Collapse>
      </Card>
      <Card>
        <Card.Header>
          <Accordion.Toggle as={Card.Header} variant="link" eventKey="2">
            <WrappedMessage
              messages={messages}
              id="accordionInstanceSettings"
            />
          </Accordion.Toggle>
        </Card.Header>
        <Accordion.Collapse eventKey="2">
          <Card.Body>
            <Nav defaultActiveKey="/home" className="flex-column">
              <NavLink exact to={ROUTES.Console.INSTANCE_SETTINGS_GENERAL}>
                <WrappedMessage messages={messages} id="linkGeneral" />
              </NavLink>
              <NavLink
                exact
                to=""
                className="disabled"
                onClick={(e) => {
                  e.preventDefault();
                }}
              >
                <WrappedMessage messages={messages} id="linkDomain" />
              </NavLink>
            </Nav>
          </Card.Body>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};
