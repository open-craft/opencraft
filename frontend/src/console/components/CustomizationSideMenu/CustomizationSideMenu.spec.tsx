import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { CustomizationSideMenu } from './CustomizationSideMenu';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<CustomizationSideMenu />).toJSON();
    expect(tree).toMatchSnapshot();
});
