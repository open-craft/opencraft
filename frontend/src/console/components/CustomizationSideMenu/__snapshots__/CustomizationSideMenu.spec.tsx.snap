// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Side Menu Page Renders with correct accordion expanded, and with current page highlighted 1`] = `
<div
  className="customization-menu accordion"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-header"
        onClick={[Function]}
        variant="link"
      >
        Theme
      </div>
    </div>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="card-body"
      >
        <div
          className="flex-column nav card-header-undefined"
          onKeyDown={[Function]}
        >
          <a
            aria-current={null}
            href="/console/theming/preview-and-colors"
            onClick={[Function]}
          >
            Preview & colors
          </a>
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Logos
          </a>
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Buttons
          </a>
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Navigation
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-header"
        onClick={[Function]}
        variant="link"
      >
        Custom Pages
      </div>
    </div>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="card-body"
      >
        <div
          className="flex-column nav card-header-undefined"
          onKeyDown={[Function]}
        >
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            About
          </a>
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Terms of Service
          </a>
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Contact
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-header"
        onClick={[Function]}
        variant="link"
      >
        Instance Settings
      </div>
    </div>
    <div
      aria-expanded={null}
      className="collapse show"
    >
      <div
        className="card-body"
      >
        <div
          className="flex-column nav card-header-undefined"
          onKeyDown={[Function]}
        >
          <a
            aria-current={null}
            href="/console/settings/general"
            onClick={[Function]}
          >
            General
          </a>
          <a
            aria-current="page"
            className="disabled active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Domain
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
